packages/embed/src/factories/create-popover/notification-days.spec.ts

```diff
@@ -0,0 +1,55 @@
+import { canBuildNotificationDot } from './notification-days'
+
+const setItemMock = jest.fn()
+
+const localStorageMockhelper = (value: any) => {
+  Object.defineProperty(window, 'localStorage', {
+    value: {
+      getItem: jest.fn().mockReturnValue(JSON.stringify(value)),
+      setItem: setItemMock,
+    },
+  })
+}
+
+describe('notification-days', () => {
+  describe('#canBuildNotificationDot', () => {
+    it('should return true when expiry date is not set in local storage and set it in local storage', () => {
+      const data = {
+        otherFormId: {
+          expiryDate: 10,
+        },
+      }
+      localStorageMockhelper(data)
+
+      expect(canBuildNotificationDot('myFormId', 2)).toBeTruthy()
+      expect(setItemMock.mock.calls[0][0]).toBe('tfNotificationData')
+      expect(setItemMock.mock.calls[0][1]).toMatch(/{"otherFormId":{"expiryDate":10},"myFormId":{"expiryDate":[0-9]+}}/)
```

Comment:
not really sure about this one
```suggestion
      expect(setItemMock).toHaveBeenCalledWith(expect.stringMatching(/{"otherFormId:{"expiryDate":10},"myFormId":{"expiryDate":[0-9]+}/)
```

Reply:
Is the other approach better? Jest docs suggest my approach: https://jestjs.io/docs/mock-functions#using-a-mock-function